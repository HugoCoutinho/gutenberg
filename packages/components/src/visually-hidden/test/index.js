/**
 * External dependencies
 */
import TestRenderer from 'react-test-renderer';

/**
 * Internal dependencies
 */
import VisuallyHidden from '../';

describe( 'VisuallyHidden', () => {
	it( 'by default renders as a div with the visually hidden class name', () => {
		const renderer = TestRenderer.create(
			<VisuallyHidden>Test</VisuallyHidden>
		);

		expect( renderer.toJSON() ).toMatchSnapshot();
	} );

	it( 'can be rendered as a different element using the `as` prop', () => {
		const renderer = TestRenderer.create(
			<VisuallyHidden as="span">Test</VisuallyHidden>
		);

		expect( renderer.toJSON() ).toMatchSnapshot();
	} );

	it( 'forwards any additional props passed to the component to the rendered element', () => {
		const renderer = TestRenderer.create(
			<VisuallyHidden as="span" data-test="test">
				Test
			</VisuallyHidden>
		);

		expect( renderer.toJSON() ).toMatchSnapshot();
	} );

	it( 'allows additional classnames to be specified, but retains the `components-visually-hidden` classname', () => {
		const renderer = TestRenderer.create(
			<VisuallyHidden as="span" className="test-component">
				Test
			</VisuallyHidden>
		);

		expect( renderer.toJSON() ).toMatchSnapshot();
	} );
} );
