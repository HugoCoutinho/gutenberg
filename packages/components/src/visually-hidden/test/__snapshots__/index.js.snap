// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VisuallyHidden allows additional classnames to be specified, but retains the \`components-visually-hidden\` classname 1`] = `
<span
  className="components-visually-hidden test-component"
>
  Test
</span>
`;

exports[`VisuallyHidden by default renders as a div with the visually hidden class name 1`] = `
<div
  className="components-visually-hidden"
>
  Test
</div>
`;

exports[`VisuallyHidden can be rendered as a different element using the \`as\` prop 1`] = `
<span
  className="components-visually-hidden"
>
  Test
</span>
`;

exports[`VisuallyHidden forwards any additional props passed to the component to the rendered element 1`] = `
<span
  className="components-visually-hidden"
  data-test="test"
>
  Test
</span>
`;
